web/client/src/api/instance.ts

```diff
@@ -126,8 +123,16 @@ function toRequestBody(obj: unknown): BodyInit {
   }
 }
 
-export function getUrlPrefix(): string {
-  return isStringEmptyOrNil(window.__BASE_URL__) ? '/' : window.__BASE_URL__
+export function getUrlWithPrefix(url: string = '/'): string {
+  let urlWithPrefix = `${
+    isStringEmptyOrNil(window.__BASE_URL__) ? '/' : window.__BASE_URL__
+  }/${url}`
+
+  while (urlWithPrefix.includes('//')) {
+    urlWithPrefix = urlWithPrefix.replaceAll('//', '/')
+  }
```

Comment:
What input are you expecting where you might need a while loop here? or is it just in case?

Reply:
it is combining `base url` with argument `url` and  both may have trailing/leading `/` or actually be a root path `/` and in that case string becomes `///`